### PR TITLE
added in chicago elec dist

### PR DESCRIPTION
**Pull request by bflatau**
Added in Chicago Elec. Dist.

### CODE REVIEW

File: lib/nerves_metal_detector/vendors/chicago_elec_dist_us.ex [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+  @behaviour Vendor
+
+  @impl Vendor
+  def vendor_info() do
+    %Vendor{
+      id: "chicagoelecdistus",
+      name: "Chicago Elec. Dist.",
+      country: :us,
+      homepage: "https://chicagodist.com/"
```
Comment (PhillippOhlandt) [inline lib/nerves_metal_detector/vendors/chicago_elec_dist_us.ex]: Can you remove the trailing `/`. (Damn this OCD all the time :D)

**Comment by PhillippOhlandt** [comment]
The parsing looks good. 

Two things:

1. Could you change the vendor module name to `ChicagoElectronicDistributorsUs` (module and file names) but keep the shorter version for the vendor `name` field?

2. Products that are in stock show a count on how many items are left, can you parse this as well? It's the same as for #30 

Product that is on stock which has an items count: https://chicagodist.com/products/12-vdc-1000ma-regulated-switching-power-adapter-ul-listed

File: lib/nerves_metal_detector/vendors/chicago_elec_dist_us.ex (hunk 1 above)
Comment (bflatau) [reply lib/nerves_metal_detector/vendors/chicago_elec_dist_us.ex]: Ah, sorry!

**Comment by bflatau** [comment]
1. I changed the names, I hope that's what you were after? 
2. I implemented a stock counter. I'm sure there's a better, more "elixir way" to do the syntax...
- I may be missing a potential error state, I'm only checking for != [] at the moment. I figure if I'm calling Floki, it will always spit out a list vs giving nil or some other value/data? 
- I'm not sure if I picked the right HTML element, but I liked that I could grab a direct value vs. doing some sort of text-to-number parsing function for the innerText since it contains both the number we want plus some text.
- Let me know what you think!

File: lib/nerves_metal_detector/inventory/data/product_update_items/chicago_electronic_distributors_us.ex [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  alias NervesMetalDetector.Vendors.ChicagoElectronicDistributorsUs.ProductUpdate
+
+  @items [
+    # RPi 4
+    %ProductUpdate{
+      url: "https://chicagodist.com/products/raspberry-pi-4-model-b-2gb",
+      sku: "RPI4-MODBP-2GB"
+    },
+    %ProductUpdate{
+      url: "https://chicagodist.com/collections/raspberry-pi/products/raspberry-pi-4-model-b-1gb",
```
Comment (PhillippOhlandt) [inline lib/nerves_metal_detector/inventory/data/product_update_items/chicago_electronic_distributors_us.ex]: Lets change the url to `https://chicagodist.com/products/raspberry-pi-4-model-b-1gb` to be consistent with the other urls.

File: lib/nerves_metal_detector/vendors/chicago_electronic_distributors_us.ex [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      Floki.find(html_tree, ".product_form") |> Floki.attribute("data-variant-inventory")
+
+    if available_quantity != [] do
+      json_product_data =
+        Floki.find(html_tree, ".product_form")
+        |> Floki.attribute("data-variant-inventory")
+        |> Enum.at(0)
+        |> Jason.decode()
+
+      get_in(elem(json_product_data, 1) |> Enum.at(0), ["inventory_quantity"])
```
Comment (PhillippOhlandt) [inline lib/nerves_metal_detector/vendors/chicago_electronic_distributors_us.ex]: You can change the code to this to reuse the existing variable and also simplify the last line where you get the value out of the json.
```elixir
        available_quantity
        |> Enum.at(0)
        |> Jason.decode!()

      Enum.at(json_product_data, 0)["inventory_quantity"]
```

File: lib/nerves_metal_detector/inventory/data/product_update_items/chicago_electronic_distributors_us.ex (hunk 2 above)
Comment (bflatau) [reply lib/nerves_metal_detector/inventory/data/product_update_items/chicago_electronic_distributors_us.ex]: Fixed! I put them in order too (to help with the OCD ha)

File: lib/nerves_metal_detector/vendors/chicago_electronic_distributors_us.ex (hunk 3 above)
Comment (bflatau) [reply lib/nerves_metal_detector/vendors/chicago_electronic_distributors_us.ex]: done!

**Comment by PhillippOhlandt** [comment]
Looks good to me! Thanks!
closes #29

**Pull request merged**